src/web/app/pages-instructor/instructor-course-enroll-page/instructor-course-enroll-page.component.ts

```diff
@@ -99,54 +101,83 @@ export class InstructorCourseEnrollPageComponent implements OnInit {
-    }, (resp: ErrorMessageOutput) => {
-      this.enrollErrorMessage = resp.error.message;
-    }, () => {
-      this.studentService.getStudentsFromCourse({ courseId: this.courseId }).subscribe((resp: Students) => {
-        this.existingStudents = resp.students;
-        if (!this.isExistingStudentsPanelCollapsed) {
-          const existingStudentTable: Handsontable = this.hotRegisterer.getInstance(this.existingStudentsHOT);
-          this.loadExistingStudentsData(existingStudentTable, this.existingStudents);
-        }
-        this.isExistingStudentsPresent = true;
-      });
-    });
+        .forEach((row: string[]) => {
+          currentStudentChunk.push({
+            section: row[hotInstanceColHeaders.indexOf(this.colHeaders[0])] === null ?
+                '' : row[hotInstanceColHeaders.indexOf(this.colHeaders[0])].trim(),
+            team: row[hotInstanceColHeaders.indexOf(this.colHeaders[1])] === null ?
+                '' : row[hotInstanceColHeaders.indexOf(this.colHeaders[1])].trim(),
+            name: row[hotInstanceColHeaders.indexOf(this.colHeaders[2])] === null ?
+                '' : row[hotInstanceColHeaders.indexOf(this.colHeaders[2])].trim(),
+            email: row[hotInstanceColHeaders.indexOf(this.colHeaders[3])] === null ?
+                '' : row[hotInstanceColHeaders.indexOf(this.colHeaders[3])].trim(),
+            comments: row[hotInstanceColHeaders.indexOf(this.colHeaders[4])] === null ?
+                '' : row[hotInstanceColHeaders.indexOf(this.colHeaders[4])].trim(),
+          });
+          if (currentStudentChunk.length >= numberOfStudentsPerRequest) {
+            allStudentChunks.push(currentStudentChunk);
+            currentStudentChunk = [];
+          }
+        });
+    if (currentStudentChunk.length > 0) {
+      allStudentChunks.push(currentStudentChunk);
+      currentStudentChunk = [];
+    }
+
+    const enrolledStudents: Student[] = [];
+
+    concat(
+        ...allStudentChunks.map((studentChunk: StudentEnrollRequest[]) => {
+          const studentsEnrollRequest: StudentsEnrollRequest = {
+            studentEnrollRequests: studentChunk,
+          };
+          return this.studentService.enrollStudents(
+              this.courseId, studentsEnrollRequest,
+          );
+        }),
+    ).pipe(finalize(() => this.isEnrolling = false))
+        .pipe(takeWhile(() => this.isEnrolling))
+        .subscribe({
+          next: (resp: Students) => {
+            enrolledStudents.push(...resp.students);
+          },
+          complete: () => {
+            this.showEnrollResults = true;
+            this.statusMessage.pop(); // removes any existing error status message
+            this.statusMessageService.showSuccessToast('Enrollment successful. Summary given below.');
+
+            const allStudentEntries: StudentEnrollRequest[] = [];
+            allStudentChunks.forEach((chunk: StudentEnrollRequest[]) => allStudentEntries.push(...chunk));
```

Comment:
Some different thoughts about this line: should we just simply construct the `allStudentEntries` at the time we construct student chunks? Would that be cleaner than having to destructure the chunks in `allStudentChunks`?

Reply:
I just think that would not make much diff since destructuring only takes 2 lines of code

Reply:
Effectively probably not. These two lines just stand out to me as an "after thought" of constructing the student list after splitting chunks. Nothing major.